### DISCUSSION TITLE
Custom component issue

### DISCUSSION THREAD

**Discussion by gabmmontenegro** [react-hook-form/react-hook-form#3015] [Q&A] [answered]
Hi everybody,

I would appreciate any help here. 

I am migrating my app from formik to react-hook-form, but I am facing an issue with a custom component. 

The component shows up six separate input fields that represents a two-factor authentication token input. Beyond the 6 input fields, the component have a seventh hidden field that is registered with react-hook-form `register` through `inputRef` prop. 

The problem is that yup isn't receiving any change event to dispatch its field validation when the 6 input fields changes the hidden input field value through a useRef. I created this codesandbox to demonstrate the issue, just enable its console.

https://codesandbox.io/s/exciting-oskar-xj4ll

The only way for yup to receive the change events is by changing directly the hidden field - which in this demonstration is visible. I also tried to manually dispatch a change event to the registered input, or by using react-hook-form `Controller` to handle the custom component, but nothing changed. 

Any idea how to solve this?

PS: With formik I was using `useField` hook.

Thanks in advance.
Gab

**Comment by bluebill1049** [comment]
what about just set use custom register at useEffect

```
useEffect(() => register('code'))

handleChange= () => { setValue('code', 'xxx') }
```

**Reply by gabmmontenegro** [reply]
Thanks for your reply, @bluebill1049.

This time I am trying to keep the component decoupled from any form library. 

Anyway, I implemented your suggestion and it worked when passing the useForm hook functions via props to the custom component. Provisionally I'll keep this solution, but seems not an ideal implementation due to dependency coupling.

Codesandbox fork with your solution: https://codesandbox.io/s/stoic-hermann-ygfk2

If any other ideas, please let me know. Thank you!

**Reply by bluebill1049** [reply]
hey @gabmmontenegro i think it's possible to decouple with form lib, but you probably want to build that component isolated and using Controller just to collect data and apply validation.

**Comment by gabmmontenegro** [comment, marked as answer]
Yay! I found out a solution to keep the component totally decoupled. Follows the codesandbox fork: https://codesandbox.io/s/vigilant-ride-6x218

Thanks @bluebill1049

**Reply by bluebill1049** [reply]
good on you 👍

**Reply by gabmmontenegro** [reply]
Taking the opportunity, what a amazing library "react-hook-form" is. Certainly a big win for the community. Thanks a lot for you and all maintainers.

**Reply by bluebill1049** [reply]
Our pleasure @gabmmontenegro thank you so much for your support too.
